torch/_dynamo/guards.py

```diff
@@ -714,10 +714,6 @@ def _parse_symbolic_shape_expressions(self, tensor_check_names, tensor_check_ids
 
         guard_expression = self.output_graph.shape_env.get_guard_expr()
         expr_as_str = guard_printer.doprint(guard_expression)
-        # We may get into a state where symbolic shape keys (all should be found in replacements)
-        # Have not been removed from the expression. This is a serious enough error state that we need to assert.
-        for key in self.output_graph.shape_env.var_to_val.keys():
-            assert str(key) not in expr_as_str, f"Unknown shape symbol {key}. "
```

Comment:
This has outlived its usefulness, since we now have this making false positives due to stuff like tensors named `vals2` triggering for `s2`